src/Compilers/CSharp/Portable/BoundTree/OutDeconstructVarPendingInference.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class OutDeconstructVarPendingInference
+    {
+        public BoundDeconstructValuePlaceholder Placeholder;
+
+        public BoundDeconstructValuePlaceholder SetInferredType(TypeSymbol type, bool success)
```

Comment:
Coercion of the arguments should create new nodes, not modify existing nodes.

---
In reply to: [66837824](https://github.com/dotnet/roslyn/pull/11873#discussion_r66837824) [](ancestors = 66837824)